src/Rel8/Expr/Opaleye.hs

```diff
@@ -39,6 +41,23 @@ castExpr :: Sql DBType a => Expr a -> Expr a
 castExpr = scastExpr typeInformation
 
 
+-- | Change the type of an 'Expr' without using @CAST()@, if the
+-- Haskell type can be safely coerced.
```

Comment:
I'm not sure if it follows that if the Haskell types can be safely coerced then the DB types can be safely coerced. Could you not have a newtype which uses a different DBType?

Reply:
I think it's sound only if you ever `newtype`-derive `DBType` instances. If that's true then it might not pull its weight: either it'll sound safer than it is, or we'll have to call it something like `sometimesButNotAlwaysSafeCoerceExpr`. Or use some constraint that indicates that it's specifically a newtype and not just `Coercible`. (e.g. the one in `generic-data`.)

Reply:
`sometimesButNotAlwaysSafe` is exactly what `unsafe` means though. `unsafe` just means that there is some pre-condition that we couldn't express in the types so you should be careful when using it

Reply:
Yeah that's my point. It's slightly less unsafe but still unsafe, which makes me doubt if it's enough of an improvement over `unsafeCoerceExpr` to include. I might just close this PR.